apis/v1alpha1/gateway_types.go

```diff
@@ -483,47 +485,125 @@ type GatewayStatus struct {
+	// constants so that operators and tools can converge on a common
+	// vocabulary to describe Gateway state.
+	//
+	// Known condition types are:
+	//
+	// * "Scheduled"
+	// * "Ready"
+	//
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	// +kubebuilder:default={{type: "Scheduled", status: "False", reason:"NotReconciled", message:"Waiting for controller", lastTransitionTime: "1970-01-01T00:00:00Z"}}
 	Conditions []metav1.Condition `json:"conditions,omitempty"`
 
 	// Listeners provide status for each unique listener port defined in the Spec.
+	//
 	// +optional
 	Listeners []ListenerStatus `json:"listeners,omitempty"`
 }
 
 // GatewayConditionType is a type of condition associated with a
 // Gateway. This type should be used with the GatewayStatus.Conditions
 // field.
+//
 type GatewayConditionType string
 
-const (
-	// ConditionForbiddenNamespaceForClass indicates that this Gateway is in
-	// a namespace forbidden by the GatewayClass.
-	ConditionForbiddenNamespaceForClass GatewayConditionType = "ForbiddenNamespaceForClass"
+// GatewayConditionReason defines the set of reasons that explain
+// why a particular Gateway condition type has been raised.
+type GatewayConditionReason string
 
+const (
 	// ConditionGatewayNotScheduled indicates that the Gateway has not been
 	// scheduled.
-	ConditionGatewayNotScheduled GatewayConditionType = "GatewayNotScheduled"
+	ConditionGatewayNotScheduled GatewayConditionType = "NotScheduled"
 
 	// ConditionListenersNotReady indicates that at least one of the specified
 	// listeners is not ready. If this condition has a status of True, a more
 	// detailed ListenerCondition should be present in the corresponding
 	// ListenerStatus.
 	ConditionListenersNotReady GatewayConditionType = "ListenersNotReady"
+)
 
-	// ConditionInvalidListeners indicates that at least one of the specified
-	// listeners is invalid. If this condition has a status of True, a more
-	// detailed ListenerCondition should be present in the corresponding
-	// ListenerStatus.
-	ConditionInvalidListeners GatewayConditionType = "InvalidListeners"
+const (
+	// GatewayConditionScheduled indicates whether the controller
+	// managing the Gateway has scheduled the Gateway to the
+	// underlying network infrastructure.
+	//
+	// Valid reasons for this condition to be false are:
+	//
+	// * "NotReconciled"
+	// * "NamespaceForbidden"
+	// * "NoSuchGatewayClass"
+	// * "NoResources"
+	GatewayConditionScheduled GatewayConditionType = "Scheduled"
+
+	// GatewayReasonNotReconciled is used when the Gateway is
+	// not scheduled because it recently been created and no
+	// controller has reconciled it yet.
+	GatewayReasonNotReconciled GatewayConditionReason = "NotReconciled"
+
+	// GatewayReasonNamespaceForbidden is used when the Gateway is
+	// not scheduled because, while the Gateway is visible to
+	// the managing controller, the controller's configuration
```

Comment:
I think it might be simpler to either not mention the Gateway being visible to the managing controller or save that for a second sentence.
```suggestion
	// not scheduled because the controller's configuration
```